File: documentation/user/en/use/api/example/evita-query-example.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => {
		return session.QuerySealedEntity(
			Query(
				Collection("Product"),
				FilterBy(
					And(
						HierarchyWithin(
							"categories",
							AttributeEquals("url", "/local-food")
						),
						EntityLocaleEquals(new CultureInfo("cs")),
						PriceValidInNow(),
						PriceInCurrency(new Currency("CZK")),
						PriceInPriceLists("vip", "loyal-customer", "regular-prices"),
						UserFilter(
							FacetHaving(
								"parameterValues",
								EntityHaving(
									AttributeInSet("code", "gluten-free", "original-recipe")
								)
							),
							PriceBetween(600m, 1600m)
						)
					)
				),
				Require(
					Page(1, 20),
					FacetSummary(Impact),
					PriceType(WithTax),
					PriceHistogram(30)
				)
			)
		);
	}
);

File: documentation/user/en/use/api/example/evita-query-example.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					and(
						hierarchyWithin(
							"categories",
							attributeEquals("url", "/local-food")
						),
						entityLocaleEquals(Locale.forLanguageTag("cs")),
						priceValidInNow(),
						priceInCurrency(Currency.getInstance("CZK")),
						priceInPriceLists("vip", "loyal-customer", "regular-prices"),
						userFilter(
							facetHaving(
								"parameterValues",
								entityHaving(
									attributeInSet("code", "gluten-free", "original-recipe")
								)
							),
							priceBetween(new BigDecimal("600"), new BigDecimal("1600"))
						)
					)
				),
				require(
					page(1, 20),
					facetSummary(IMPACT),
					priceType(WITH_TAX),
					priceHistogram(30, STANDARD)
				)
			)
		);
	}
);